pydantic/types.py

```diff
@@ -626,3 +634,238 @@ def display(self) -> str:
+        """
+        Main logic of color parsing
+        """
+        if isinstance(self._original, tuple):
+            self._color_match = self._parse_tuple(self._original)
+
+        elif isinstance(self._original, str):
+            color = self._original.lower()
+
+            # rgb/rgba string
+            self._color_match = self._parse_rgb_str(color)
+            if self._color_match is not None:
+                return
+
+            # hex string
+            self._color_match = self._parse_hex_str(color)
+            if self._color_match is not None:
+                return
+
+            # named colour
+            if color in colors.BY_NAME:
+                self._color_match = color
+                return
+
+    def _get_color_or_raise(self) -> None:
+        """
+        Raise error if color name not found
+        """
+        if self._color_match is None:
+            raise errors.ColorError()
+
+    def original(self) -> ColorType:
+        """
+        Return original value passed to the model
+        """
+        return self._original
+
+    def as_hex(self, prefix: str = '0x', reduce: bool = True) -> str:
+        """
+        Return hexadecimal value of the color
+
+        Try reduce hex code to 3-digit format, fallback to 6-digit.
+        """
+        self._get_color_or_raise()
+        hexadecimal, _ = colors.BY_NAME[self._color_match or '']
+        return '{}{}'.format(prefix, self.reduce_6_digit_hex(hexadecimal) if reduce else hexadecimal)
+
+    def as_tuple(self, alpha: bool = False) -> AnyRGBType:
+        """
+        Return RGB or RGBA tuple
+        """
+        self._get_color_or_raise()
+        _, rgb = colors.BY_NAME[self._color_match or '']
+        r, g, b = rgb
+
+        if alpha:
+            return r, g, b, 1.0
+        return r, g, b
+
+    def as_rgb(self, alpha: bool = False) -> str:
+        """
+        Return RGB or RGBA string representation of the color
+        """
+        rgb = self.as_tuple(alpha)
+        if alpha:
+            return 'rgba({}, {}, {}, {})'.format(*rgb)
+        return 'rgb({}, {}, {})'.format(*rgb)
+
+    def as_named_color(self) -> str:
+        """
+        Return name of the color as per CSS3 specification.
+        """
+        self._get_color_or_raise()
+        return self._color_match  # type: ignore
+
+    def as_hls(self) -> HLSType:
+        """
+        Return tuple of floats representing Hue Lightness Saturation (HLS) color
+        """
+
+        def normalize(v: Union[int, float, str]) -> float:
+            return float(v) / 255
+
+        self._get_color_or_raise()
+        r, g, b = self.as_tuple(alpha=False)  # type: ignore
+        return rgb_to_hls(normalize(r), normalize(g), normalize(b))
+
+    def __str__(self) -> str:
+        return str(self._original).lower()
+
+    def __repr__(self) -> str:
+        return '<Color("{}")>'.format(self._color_match or self._original)
+
+    @staticmethod
+    def dumps(value: 'Color') -> str:
```

Comment:
surely we should just use `str` here, that's what people would expect.

Reply:
I removed this method, as it only needed for as JSON encoder. I've rewritten encoder in ``json`` module as a lambda expression